app/src/main/java/com/ucm/informatica/spread/Activities/MainTabActivity.java

```diff
@@ -11,15 +12,29 @@
 import android.widget.RelativeLayout;
 import android.widget.TextView;
 
-import com.ucm.informatica.spread.Contracts.NameContract;
+import com.andrognito.flashbar.Flashbar;
+import com.andrognito.flashbar.anim.FlashAnim;
+import com.mapbox.geojson.Point;
+import com.ucm.informatica.spread.Contracts.CoordContract;
+import com.ucm.informatica.spread.Model.Event;
+import com.ucm.informatica.spread.Model.Region;
 import com.ucm.informatica.spread.Presenter.MainTabPresenter;
 import com.ucm.informatica.spread.R;
-import com.ucm.informatica.spread.SmartContract;
+import com.ucm.informatica.spread.Utils.SmartContract;
 import com.ucm.informatica.spread.View.MainTabView;
-import com.ucm.informatica.spread.ViewPagerAdapter;
-import com.ucm.informatica.spread.ViewPagerTab;
+import com.ucm.informatica.spread.Utils.ViewPagerAdapter;
+import com.ucm.informatica.spread.Utils.ViewPagerTab;
 
-import static com.ucm.informatica.spread.Constants.NUMBER_TABS;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import timber.log.Timber;
+import static com.ucm.informatica.spread.Utils.Constants.NUMBER_TABS;
 
 public class MainTabActivity extends AppCompatActivity implements MainTabView{
 
```

Comment:
main activity get and set data from outsider storage (blockchain and txt file with coords)